### PR TITLE
HDDS-4385. Search function using container ID on Recon Missing Container

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
To make Container ID Column as searchable.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4385

## How was this patch tested?
Manually

image

### CODE REVIEW

**Comment by kerneltime** [comment]
@GeorgeJahad

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx [hunk 1]
```diff
@@ -355,28 +357,80 @@ export class MissingContainers extends React.Component<Record<string, object>, I
...
             </TabPane>
             <TabPane key='3' tab='Over-Replicated'>
               <Table
-                expandRowByClick dataSource={overReplicatedDataSource} columns={CONTAINER_TAB_COLUMNS}
+                expandRowByClick dataSource={overReplicatedDataSource}
+                columns={CONTAINER_TAB_COLUMNS.reduce<any[]>((filtered, column) => {
+                  if (column.isSearchable) {
+                    const newColumn = {
+                      ...column,
+                      ...new ColumnSearch(column).getColumnSearchProps(column.dataIndex)
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:405]: just curious~ if in the future want to enable other columns to be searchable,
do we have to _new ColumnSearch_ object for each column here as well?

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:405]: Just  need to use this isSearchable: true line in columns.

Comment (DaveTeng0) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:405]: Got it! Thanks! And it'd be nice (maybe in another jira) if we could pull out the common parts between Missing, Under-Replicated, Over-Replicated, and Mis-Replicated and reuse!

**Comment by GeorgeJahad** [comment]
lgtm, but as @DaveTeng0 mentioned it would nice to reduce the code duplication.  (I should also mention that I haven't written much typescript.)

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx (hunk 1 above)
Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:405]: I agree with @DaveTeng0, as I checked the only difference between the `Table`s inside the `TabPane`s are the `dataSource`. Maybe we could create a method with one parameter (`dataSource`) inside this file which creates this `Table` with the given `dataSource`.

**Review by dombizita — COMMENTED** [review]
Thanks for your patch @smitajoshi12, it looks good to me, but maybe we could simplify the code with the suggestion in the inline comments.

**Comment by smitajoshi12** [comment]
> Thanks for your patch @smitajoshi12, it looks good to me, but maybe we could simplify the code with the suggestion in the inline comments.

 Done Changes as Suggested by Zita

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx [hunk 2]
```diff
@@ -341,6 +341,22 @@ export class MissingContainers extends React.Component<Record<string, object>, I
     return <div>Loading...</div>;
   };
 
+  serachCoulmn = () => {
```
Comment (GeorgeJahad) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:344]: searchColumn?

Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:344]: Couldn't we move the whole `Table` into the new method? Something like this:
```
generateTable = (dataSource: IContainerResponse[]) => {
  <Table
    expandRowByClick dataSource={dataSource}
    columns={CONTAINER_TAB_COLUMNS.reduce<any[]>((filtered, column) => {
      if (column.isSearchable) {
        const newColumn = {
          ...column,
          ...new ColumnSearch(column).getColumnSearchProps(column.dataIndex)
        };
        filtered.push(newColumn);
      } else {
        filtered.push(column);
      }
  
      return filtered;
    }, [])}
    loading={loading}
    pagination={paginationConfig} rowKey='containerID'
    expandedRowRender={this.expandedRowRender} onExpand={this.onRowExpandClick}/>
}
```

**Review by smengcl — APPROVED** [review]
Thanks @smitajoshi12 for the patch. Thanks @GeorgeJahad and @dombizita for the review.

**Pull request merged**